mlflow/utils/environment.py

```diff
@@ -18,11 +26,131 @@
 _CONDA_ENV_FILE_NAME = "conda.yaml"
 _REQUIREMENTS_FILE_NAME = "requirements.txt"
 _CONSTRAINTS_FILE_NAME = "constraints.txt"
+_PYTHON_ENV_FILE_NAME = "python-env.yaml"
+
+
+_PYTHON_REGEX = re.compile(r"^python(=|==|<=)([\d.]+)$")  # matches, e.g., "python=3.7.12"
+_BUILD_PACKAGE_REGEX = re.compile(
+    r"^(pip|setuptools|wheel)(<|>|<=|>=|=|==|!=)([\d.]+)$"
+)  # matches, e.g., "pip=22.0.4"
+
+
+class PythonEnv:
+    """
+    Represents environment information for an MLflow Model.
+    """
+
+    def __init__(self, python=None, build_dependencies=None, dependencies=None):
+        if python is not None and not isinstance(python, str):
+            raise TypeError(f"`python` must be a string but got {type(python)}")
+        if build_dependencies is not None and not isinstance(build_dependencies, list):
+            raise TypeError(
+                f"`build_dependencies` must be a list but got {type(build_dependencies)}"
+            )
+        if dependencies is not None and not isinstance(dependencies, list):
+            raise TypeError(f"`dependencies` must be a list but got {type(dependencies)}")
+
+        self.python = python
+        self.build_dependencies = build_dependencies
+        self.dependencies = dependencies
+
+    def __str__(self):
+        return str(self.to_dict())
+
+    @classmethod
+    def default(cls):
+        return cls(
+            python=cls._get_current_python(),
+            build_dependencies=cls.get_default_build_dependencies(),
+            dependencies=[f"-r {_REQUIREMENTS_FILE_NAME}"],
+        )
```

Comment:
Looking for a better name for this class method.

Reply:
This method is used in a flavor to provide the default python env config.

Reply:
> Looking for a better name for this class method.

Default is good :)

Reply:
Makes sense. I'll rename it.